byterun/pyvm2.py

```diff
@@ -166,18 +167,36 @@ def unwind_block(self, block):
 
     def parse_byte_and_args(self):
         """ Parse 1 - 3 bytes of bytecode into
-        an instruction and optionally arguments."""
+        an instruction and optionally arguments.
+        In Python3.6 the format is 2 bytes per instruction."""
         f = self.frame
         opoffset = f.f_lasti
-        byteCode = byteint(f.f_code.co_code[opoffset])
+        if f.py36_opcodes:
+            currentOp = f.py36_opcodes[opoffset]
+            byteCode = currentOp.opcode
+            byteName = currentOp.opname
+        else:
+            byteCode = byteint(f.f_code.co_code[opoffset])
+            byteName = dis.opname[byteCode]
         f.f_lasti += 1
-        byteName = dis.opname[byteCode]
         arg = None
         arguments = []
+        if f.py36_opcodes and byteCode == dis.EXTENDED_ARG:
+            # Prefixes any opcode which has an argument too big to fit into the
+            # default two bytes. ext holds two additional bytes which, taken
+            # together with the subsequent opcode’s argument, comprise a
+            # four-byte argument, ext being the two most-significant bytes.
+            # We simply ignore the EXTENDED_ARG because that calculation
+            # is already done by dis, and stored in next currentOp.
+            # Lib/dis.py:_unpack_opargs
+            return self.parse_byte_and_args()
         if byteCode >= dis.HAVE_ARGUMENT:
-            arg = f.f_code.co_code[f.f_lasti:f.f_lasti+2]
-            f.f_lasti += 2
-            intArg = byteint(arg[0]) + (byteint(arg[1]) << 8)
+            if f.py36_opcodes:
+                intArg = currentOp.arg
+            else:
```

Comment:
also here, I am not sure why we are handling 3.6 in such a special way, the code below looks correct in either case.

Reply:
I will update to look for `3.4` instead.